`TestCreateSignedSTSIdentityRequest` flakiness

```
=== Failed
=== FAIL: lib/auth/join/iam TestCreateSignedSTSIdentityRequest/us-west-2_fips_with_region_from_imdsv2 (0.00s)
    iam_test.go:159:
                Error Trace:    /src/teleport/lib/auth/join/iam/iam_test.go:159
                Error:          Not equal:
                                expected: "sts-fips.us-west-2.amazonaws.com"
                                actual  : "sts-fips.us-east-1.amazonaws.com"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -sts-fips.us-west-2.amazonaws.com
                                +sts-fips.us-east-1.amazonaws.com
                Test:           TestCreateSignedSTSIdentityRequest/us-west-2_fips_with_region_from_imdsv2
    iam_test.go:165:
                Error Trace:    /src/teleport/lib/auth/join/iam/iam_test.go:165
                Error:          Not equal:
                                expected: "us-west-2"
                                actual  : "us-east-1"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -us-west-2
                                +us-east-1
                Test:           TestCreateSignedSTSIdentityRequest/us-west-2_fips_with_region_from_imdsv2
                Messages:       signature region did not match expected
    --- FAIL: TestCreateSignedSTSIdentityRequest/us-west-2_fips_with_region_from_imdsv2 (0.00s)

=== FAIL: lib/auth/join/iam TestCreateSignedSTSIdentityRequest/us-west-2_with_region_from_imdsv2 (0.00s)
    iam_test.go:159:
                Error Trace:    /src/teleport/lib/auth/join/iam/iam_test.go:159
                Error:          Not equal:
                                expected: "sts.us-west-2.amazonaws.com"
                                actual  : "sts.us-east-1.amazonaws.com"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -sts.us-west-2.amazonaws.com
                                +sts.us-east-1.amazonaws.com
                Test:           TestCreateSignedSTSIdentityRequest/us-west-2_with_region_from_imdsv2
    iam_test.go:165:
                Error Trace:   /src/teleport/lib/auth/join/iam/iam_test.go:165
                Error:          Not equal:
                                expected: "us-west-2"
                                actual  : "us-east-1"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -us-west-2
                                +us-east-1
                Test:           TestCreateSignedSTSIdentityRequest/us-west-2_with_region_from_imdsv2
                Messages:       signature region did not match expected
    --- FAIL: TestCreateSignedSTSIdentityRequest/us-west-2_with_region_from_imdsv2 (0.00s)

=== FAIL: lib/auth/join/iam TestCreateSignedSTSIdentityRequest (0.01s)
```

--- a/lib/auth/join/iam/iam_test.go
+++ b/lib/auth/join/iam/iam_test.go
@@ -38,6 +38,10 @@ func TestCreateSignedSTSIdentityRequest(t *testing.T) {
 	t.Setenv("AWS_SECRET_ACCESS_KEY", "FAKE_KEY")
 	t.Setenv("AWS_SESSION_TOKEN", "FAKE_SESSION_TOKEN")
 
+	// If the user has an AWS config file that sets a region, it conflicts with
+	// the test cases. Disable loading a config file by setting a nonexistent path.
+	t.Setenv("AWS_CONFIG_FILE", "fake-file-this-must-not-exist")
+
 	const challenge = "asdf12345"
 
 	for desc, tc := range map[string]struct {

